awx/ui_next/src/screens/Template/WorkflowJobTemplateVisualizer/Modals/NodeModals/NodeModal.jsx

```diff
@@ -0,0 +1,213 @@
+            type: 'workflow_approval_template',
+          }
+        : nodeResource;
+
+    onSave(linkType, resource, nodeType);
+  };
+
+  const handleCancel = () => {
+    clearQueryParams();
+    dispatch({ type: 'CANCEL_NODE_MODAL' });
+  };
+
+  const handleNodeTypeChange = newNodeType => {
+    setNodeType(newNodeType);
+    setNodeResource(null);
+    setApprovalName('');
+    setApprovalDescription('');
+    setApprovalTimeout(0);
+  };
+
+  const steps = [
+    ...(askLinkType
+      ? [
+          {
+            name: i18n._(t`Run Type`),
+            key: 'run_type',
+            component: (
+              <RunStep linkType={linkType} onUpdateLinkType={setLinkType} />
+            ),
+            enableNext: linkType !== null,
+          },
+        ]
+      : []),
+    {
+      name: i18n._(t`Node Type`),
+      key: 'node_resource',
+      enableNext:
+        (nodeType !== 'approval' && nodeResource !== null) ||
+        (nodeType === 'approval' && approvalName !== ''),
+      component: (
+        <NodeTypeStep
+          description={approvalDescription}
+          name={approvalName}
+          nodeResource={nodeResource}
+          nodeType={nodeType}
+          onUpdateDescription={setApprovalDescription}
+          onUpdateName={setApprovalName}
+          onUpdateNodeResource={setNodeResource}
+          onUpdateNodeType={handleNodeTypeChange}
+          onUpdateTimeout={setApprovalTimeout}
+          timeout={approvalTimeout}
+        />
+      ),
+    },
+  ];
+
+  steps.forEach((step, n) => {
+    step.id = n + 1;
+  });
+
+  const CustomFooter = (
+    <WizardFooter>
+      <WizardContextConsumer>
+        {({ activeStep, onNext, onBack }) => (
+          <>
+            <NodeNextButton
+              triggerNext={triggerNext}
+              activeStep={activeStep}
+              onNext={onNext}
+              onClick={() => setTriggerNext(triggerNext + 1)}
+              buttonText={
+                activeStep.key === 'node_resource'
+                  ? i18n._(t`Save`)
+                  : i18n._(t`Next`)
+              }
+            />
+            {activeStep && activeStep.id !== 1 && (
+              <Button variant="secondary" onClick={onBack}>
+                {i18n._(t`Back`)}
+              </Button>
+            )}
+            <Button variant="link" onClick={handleCancel}>
+              {i18n._(t`Cancel`)}
+            </Button>
+          </>
+        )}
+      </WizardContextConsumer>
+    </WizardFooter>
+  );
+
+  const wizardTitle = nodeResource ? `${title} | ${nodeResource.name}` : title;
+
+  return (
+    <Wizard
+      footer={CustomFooter}
+      isOpen
+      onClose={handleCancel}
+      onSave={handleSaveNode}
+      steps={steps}
+      css="overflow: scroll"
+      title={wizardTitle}
+    />
+  );
+}
+
+NodeModal.propTypes = {
+  askLinkType: bool.isRequired,
+  onSave: func.isRequired,
+  title: node.isRequired,
+};
+
+export default withI18n()(withRouter(NodeModal));
```

Comment:
Instead of using `withRouter`, you could pull in react-router hooks `let history = useHistory();`.

Reply:
👍 there were a few other places where I was using withRouter.  I'll replace those as well.